### ISSUE TITLE
Unable to logout on Seneca's IdP (staging or production)

### ISSUE THREAD

**Issue by cindyledev** [Seneca-CDOT/telescope#982]
**What happened**:
I can log onto https://dev.telescope.cdot.systems/ using my senstudents.com with no problem but I cannot log out. When I click on LOGOUT, I get redirected to this:
![Annotation 2020-04-09 231359](https://user-images.githubusercontent.com/36822198/78959391-cbccf700-7ab8-11ea-88be-b160f32036c5.png)

**What should have happened**:
I should be able to log in and log out because my account is legit.

**How to reproduce it (as precise as possible)**:
Try logging in and logging out using your senstudents.com SSO account

**Anything else we need to know?**:

**Environment**:

- OS: Windows
- Browser: Chrome/Firefox

**Comment by humphd** [comment]
When I updated deps recently, we went from `passport-saml` v1.2.0 to v1.3.3, and I wonder if [something in between](https://github.com/bergie/passport-saml/compare/v1.2.0...v1.3.3) broke logout.

We could try rolling that back as an initial test.  I'd also like to see if this is reproducable locally, if you eliminate Seneca from the equation.

**Comment by humphd** [comment]
I tested locally, and can't reproduce with `v1.3.3`.  I'm going to email ITS and get their thoughts.

**Comment by humphd** [comment]
I tried downgrading passport-saml back to v1.2.0 and it made no difference on staging.

**Comment by humphd** [comment]
I worked on this a bunch today, and got us to a bit better spot, but this isn't working as well as I'd like yet.

Mehrdad suggested that with Single Logout we should try to use redirect vs post.  I did some reading and according to the passport-saml docs (see https://github.com/bergie/passport-saml#slo-single-logout) the passport-saml SLO supports:

“Redirect and POST SAML Protocol Bindings”

But I confess that I don’t understand what this means from a practical configuration standpoint for our SP.

Reading the PR that added it (https://github.com/bergie/passport-saml/commit/7674e18e2cc674132a9655936c2fca5dfc72af70) and the original Issue that was filed to add it (https://github.com/bergie/passport-saml/issues/191), it sounds like it should just work, if the IdP supports it (i.e., they will use either).

I tried a change in our code that seems like it could deal with this (see https://github.com/Seneca-CDOT/telescope/commit/59be72e1c60ad365bd6ac8c3501fcf1db65372e2) but it made no difference.

So I added one more change to have it do a logout on the SP side instead of redirecting at all to the IdP.  Now you can logout at our app, but you aren’t really logged out at the other end.

Basically it “works” but it’s not right.  I think you should be logged out fully at the IdP?

One additional thing we could try is to use `forceAuthn: true`:

> `forceAuthn`: if set to true, the initial SAML request from the service provider specifies that the IdP should force re-authentication of the user, even if they possess a valid session.

**Comment by humphd** [comment]
Fixed by #1051, #1053, #1054.  Verified fixed on staging. Closing.

We should do a 1.0.1 soon to get this on production.
